addons/udes_stock/models/stock_quant.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, _, api
+from collections import defaultdict
+
+class StockQuant(models.Model):
+    _inherit = 'stock.quant'
+
+    def _gather(self, product_id, location_id, **kwargs):
+        """ Call default _gather function, if quant_ids context variable
+            is set the resulting quants are filtered by id.
+            Context variable quant_ids might contain quants of different products.
+            TODO: Extend default _gather to include quant_ids as kwarg
+        """
+        quants = super(StockQuant, self)._gather(product_id, location_id, **kwargs)
+        quant_ids = self.env.context.get('quant_ids')
+        if quant_ids:
+            quants = quants.filtered(lambda q: q.id in quant_ids)
+        return quants
+
+    def total_quantity(self):
+        """ Returns the total quantity of the quants in self
+        """
+        return sum(self.mapped('quantity'))
+
+    def group_quantity_by_product(self, get_key=lambda p: p.product_id.id, only_available=False):
```

Comment:
Using `q` would be better for lambda (since we are dealing with quants.

Do we want `p.product_id.id` or just `p.product_id`?

Reply:
Fixed to q.product_id